### ISSUE TITLE
cargo doc is failing

### ISSUE THREAD

**Issue by oysterpack** [rust-lang-nursery/futures-rs#1497]
running `cargo +nightly doc` on a project that includes futures-preview fails with the below error:
```
...
error: `[Box::pinned]` cannot be resolved, ignoring it...
   --> /home/alfio/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-preview-0.3.0-alpha.13/src/try_stream/mod.rs:602:84
    |
602 |     /// first have to pin the stream. This can be done by boxing the stream using [`Box::pinned`]
    |                                                                                    ^^^^^^^^^^^^^ cannot be resolved, ignoring
    |
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

error: aborting due to 147 previous errors

Caused by:
  process didn't exit successfully: `rustdoc --edition=2018 --crate-name ruggine_concurrency ruggine-concurrency/src/lib.rs  ...
```
## Meta
- rustdoc 1.35.0-nightly (94fd04589 2019-03-21)

**Comment by taiki-e** [comment]
The old method name will fix in #1498, but the errors is a problem of rustdoc (see https://github.com/rust-lang/rust/issues/43466#issuecomment-410650216 and https://github.com/rust-lang/rust/issues/56922).

Also, if that project uses `#![deny(warnings)]` or `RUSTDOCFLAGS=-Dwarnings`, you may be able to avoid the error by changing it (e.g., use `#![cfg_attr(test, deny(warnings))]`, remove it, etc). (`intra_doc_link_resolution_failure` is a warning by default).

**Comment by oysterpack** [comment]
Thanks - the project uses `#![deny(warnings)]` and removing it resolved the issue ... now I can generate docs for the project excluding dependencies
